src/lib.rs

```diff
@@ -141,7 +144,7 @@ where
 
 const MAX_FREQ_KHZ: u16 = 500;
 const USECS_IN_MSEC: u16 = 1_000;
-const DELAY_USECS: u16 = USECS_IN_MSEC / MAX_FREQ_KHZ;
+const DELAY_USECS: u16 = USECS_IN_MSEC.div_ceil(MAX_FREQ_KHZ * 2);
```

Comment:
The datasheet says "the clock frequency should be less than 250K" and now I have no idea why this library works at all =) coz it sends data at 500khz speed.

So I believe we could leave the original value here then 2 delays would result in 500khz / 2 = 250khz clock frequency (like datasheet requires).

Reply:
He-he, here it was [speedup](https://github.com/igelbox/tm1637-rs/pull/1/files#r415055526)

Reply:
https://raw.githubusercontent.com/avishorp/TM1637/master/docs/TM1637_V2.4_EN.pdf specifies the maximum frequency as 500kHz. Perhaps there are different revisions of the chip with different requirements?

Reply:
And they mention "the clock frequency should be less than 250K". However it may relate to reading not writing. Not sure though.

I had just one chip. I'm not an experienced developer of embeddeds :shrug:)